app/controllers/builds_controller.rb

```diff
@@ -92,8 +92,18 @@ def edit_build_params
     params.require(:build).permit(:label, :description)
   end
 
+  def extra_docker_build_params
+    hash = {}
+    Samson::Hooks.fire(:extra_build_params).each do |p|
+      hash[p] = params[p].present?
+    end
+    hash
+  end
+
   def start_docker_build
-    DockerBuilderService.new(@build).run!(push: true)
+    docker_build_params = { push: true }
+    docker_build_params.update(extra_docker_build_params)
+    DockerBuilderService.new(@build).run!(docker_build_params)
```

Comment:
nvm ... build_params would be attributes for the build ... this is something else ...